pxtblocks/monkeyPatches/gesture.ts

```diff
@@ -0,0 +1,114 @@
+import * as Blockly from "blockly";
+
+interface PatchedGesture extends Blockly.Gesture {
+    id: number | undefined
+    boundEvents_: Blockly.browserEvents.Data[];
+}
+
+export function monkeyPatchGesture() {
+    // This monkey patch is only required for the in-game experience of Minecraft on ChromeOS.
+    // For some reason, events are occasionally dropped by the Android webview when multitouch
+    // is being used which can cause Blockly to get stuck thinking it's in a gesture when it isn't.
+    // This patch effectively removes multitouch by ignoring all pointer events except for ones
+    // generated by the first pointer that is encountered. All pointerup events are still allowed
+    // through (see wrapHandler below)
+    if (!pxt.BrowserUtils.isChromeOS() || !pxt.BrowserUtils.isInGame()) {
+        return;
+    }
+
+    const oldDoStart = Blockly.Gesture.prototype.doStart;
+    Blockly.Gesture.prototype.doStart = function(this: PatchedGesture, e: PointerEvent) {
+        if (this.id) {
+            e.stopPropagation();
+            e.preventDefault();
+            return;
+        }
+
+        this.id = e.pointerId;
+        oldDoStart.call(this, e);
+    }
+
+    Blockly.Gesture.prototype.bindMouseEvents = function (this: PatchedGesture, e: PointerEvent) {
+        if (!this.boundEvents_) this.boundEvents_ = [];
+
+        const createSyntheticEvent = (e: PointerEvent) => {
+            const syntheticEvent = new PointerEvent(e.type, {
+                ...e,
+                clientX: e.clientX,
+                clientY: e.clientY,
```

Comment:
Why to clientX and clientY have to be respecified here?

Reply:
pointer events are native objects and i was seeing some strange behavior where certain properties weren't being assigned when using the spread operator. as a safety measure, i went ahead and directly specified everything that is used by Blockly's gesture implementation today